test_app/views.py

```diff
@@ -21,14 +24,40 @@ class TeamViewSet(TestAppViewSet):
     serializer_class = serializers.TeamSerializer
 
 
-class UserViewSet(TestAppViewSet):
+class UserViewSet(ModelViewSet):
+    permission_classes = [permissions.IsAuthenticated]
     serializer_class = serializers.UserSerializer
 
+    def get_queryset(self):
+        return User.objects.all()
```

Comment:
Minor: Unless you are doing something special per-user, it makes more sense to set the `queryset` attribute on the class.

Reply:
I owe an update on this. The issue here is - `User.objects.all()` is the _wrong answer_. That says "always show all users". But awkwardly, users don't have permissions or roles in this system. In the AWX system, permission to edit users came from _organization_ permissions. But it gets worse.

Due to reasons, there really does need to be canonical rules for who can see users (this queryset, dependent on requesting user) and do the other user CRUD actions. There are legacy rules to consider, and they are not the same. They're not close to being the same. Implementing a set of rules for user CRUD would be a simple task - frankly easy. Getting people to understand that there is a conflict of rules, agree that a single set of rules is needed, and then agree on what those rules should be... is an exhausting thought. If I were to hand-wave that as done, then I would agree that it would be a validly in the domain of DAB RBAC.

Reply:
Both `.list()` and `.get_object()` on the base `ModelViewSet` class by default call `self.filter_queryset(self.get_queryset())`.  The key for permissions control is either `.filter_queryset` (for both `.list` and `.get_object`, if we want to prevent even seeing some things) or `.check_object_permissions` (specifically for `.get_object`), not in hand-crafting a starting queryset.

- https://www.cdrf.co/3.14/rest_framework.viewsets/ModelViewSet.html#get_object
- https://www.cdrf.co/3.14/rest_framework.viewsets/ModelViewSet.html#list
- https://www.cdrf.co/3.14/rest_framework.viewsets/ModelViewSet.html#filter_queryset

Reply:
We might be talking a bit at cross-purposes here.  My concern is more making use of more of the customization hooks that Django REST Framework provides, rather than overloading everything into `.get_queryset()`.

Reply:
Sure, agreed in principal on moving away from `get_queryset` to `filter_queryset`. That's no big deal. I think we do need to allow the bound method `access_qs` accept a base queryset, which is no technical problem at all.

Reply:
I looked into this more, and I just don't want to do it right now. This creates a lot of new disturbances within test_app. Could we maybe file an issue to do this later?